
/*========
        This file is part of SizzlingPlugins.

    Copyright (c) 2010-2013, Jordan Cristiano.
    This file is subject to the terms and conditions 
    defined in the file 'LICENSE', which is part of this
    source code package.
    
*/

package SizzEvent;

option optimize_for = LITE_RUNTIME;

message SizzEvent
{
	message EventData
	{
		enum DATA_TYPE
		{
			// specially named for json parsing on the web
			valueString = 1;
			valueFloat = 2;
			valueLong = 3;
			valueShort = 4;
			valueByte = 5;
			valueBool = 6;
		}
		
		optional string key_name = 1;
		
		optional DATA_TYPE value_type = 2;
		optional string value_string = 3;
		optional float value_float = 4;
		optional int32 value_long = 5;
		optional int32 value_short = 6;
		optional int32 value_byte = 7;
		optional bool value_bool = 8;
	}

	optional uint32 event_timestamp = 1;
	optional string event_name = 2;

	repeated EventData event_data = 3;
}
